azurelinuxagent/common/protocol/wire.py

```diff
@@ -161,12 +161,7 @@ def download_ext_handler_pkg(self, uri, destination, headers=None, use_proxy=Tru
         # in the lambda.
         host_func = lambda: self._download_ext_handler_pkg_through_host(uri, destination)
 
-        try:
-            success = self.client.send_request_using_appropriate_channel(direct_func, host_func)
-        except Exception:
-            success = False
-
-        return success
+        return self.client.send_request_using_appropriate_channel(direct_func, host_func)
```

Comment:
Previously download_ext_handler_pkg was returning False on error... seems like now it can return None

Reply:
This change is intentional. Now all functions getting called in `send_request_using_appropriate_channel` return None on failure.

Reply:
Discussed offline; reverted this change to keep interface consistent with current behaviour.